spec/formatting.md

```diff
@@ -219,15 +222,66 @@ the following steps are taken:
    as long as reasonable precautions are taken to keep the function interface
    simple and minimal, and avoid introducing potential security vulnerabilities.
 
-   As implementations MAY allow custom functions to be defined by users,
-   their access to the _formatting context_ SHOULD be minimal and read-only,
-   and their execution time SHOULD be limited.
+   Implementations MAY expose _options_ in an implementation-defined _namespace_
+   in addition to or superceding those found in the default registry.
+
+   An implementation MAY define its own functions.
+   An implementation MAY allow custom functions to be defined by users.
+
+   Function access to the _formatting context_ MUST be minimal and read-only,
+   and execution time SHOULD be limited.
+   Implementation-defined _functions_ SHOULD use an implementation-defined _namespace_.
+   User-defined _functions_ SHOULD be permitted to define the _namespace_ to use
+   in name resolution.
```

Comment:
Does this mean that a user can define their own namespaces (e.g. introduce new namespaces with arbitrary names), or that they can choose a namespace from a set of existing namespaces?